application/config/database.php

```diff
@@ -40,40 +39,65 @@
+|			'mysqli' and 'pdo/mysql' drivers accept an array with the following options:
+|
+|				'ssl_key'    - Path to the private key file
+|				'ssl_cert'   - Path to the public key certificate file
+|				'ssl_ca'     - Path to the certificate authority file
+|				'ssl_capath' - Path to a directory containing trusted CA certificates in PEM format
+|				'ssl_cipher' - List of *allowed* ciphers to be used for the encryption, separated by colons (':')
+|				'ssl_verify' - TRUE/FALSE; Whether verify the server certificate or not
+|
+|	['compress'] Whether or not to use client compression (MySQL only)
 |	['stricton'] TRUE/FALSE - forces 'Strict Mode' connections
 |							- good for ensuring strict SQL while developing
+|	['ssl_options']	Used to set various SSL options that can be used when making SSL connections.
+|	['failover'] array - A array with 0 or more data for connections if the main should fail.
+|	['save_queries'] TRUE/FALSE - Whether to "save" all executed queries.
+| 				NOTE: Disabling this will also effectively disable both
+| 				$this->db->last_query() and profiling of DB queries.
+| 				When you run a query, with this setting set to TRUE (default),
+| 				CodeIgniter will store the SQL statement for debugging purposes.
+| 				However, this may cause high memory usage, especially if you run
+| 				a lot of SQL queries ... disable this to avoid that problem.
 |
 | The $active_group variable lets you choose which connection group to
 | make active.  By default there is only one group (the 'default' group).
 |
-| The $active_record variables lets you determine whether or not to load
-| the active record class
+| The $query_builder variables lets you determine whether or not to load
+| the query builder class.
 */
-
 $active_group = 'default';
-$active_record = TRUE;
+$query_builder = TRUE;
 
-$db['default']['hostname'] = getenv('DB_HOST') ?: '';
-$db['default']['username'] = getenv('DB_USER') ?: '';
-$db['default']['password'] = getenv('DB_PASSWORD' ?: '');
-$db['default']['database'] = getenv('DB_NAME') ?: '';
-$db['default']['dbdriver'] = 'mysqli';
-$db['default']['dbprefix'] = '';
-$db['default']['pconnect'] = TRUE;
-$db['default']['cache_on'] = FALSE;
-$db['default']['cachedir'] = '';
-$db['default']['char_set'] = 'utf8';
-$db['default']['dbcollat'] = 'utf8_general_ci';
-$db['default']['swap_pre'] = '';
-$db['default']['autoinit'] = TRUE;
 $db['default']['stricton'] = FALSE;
 // set db_debug to FALSE in production and when testing w/o a database
 //$db['default']['db_debug'] = FALSE;
 $db['default']['db_debug'] = FALSE;
 if (filter_var(getenv('DB_DEBUG'), FILTER_VALIDATE_BOOLEAN)) {
     $db['default']['db_debug'] = TRUE;
 }
-
-/* End of file database.php */
-/* Location: ./application/config/database.php */
+$db['default'] = array(
+	'dsn'	=> '',
+	'hostname' => getenv('DB_HOST') ?: '',
+	'username' => getenv('DB_USER') ?: '',
+	'password' => getenv('DB_PASSWORD') ?: '',
+	'database' => getenv('DB_NAME') ?: '',
```

Comment:
I wonder if these env variables should be required. i.e. we explicitly fail to start if they are unset. What do you think?

Reply:
Taking care of that [over here](https://github.com/GSA/project-open-data-dashboard/blob/a81d15713c89b21cfcdbb37a438d6360dc02bfb7/application/config/database.php#L105-L111) now.